src/subdomains/generic/user/models/user/user.service.ts

```diff
@@ -240,15 +245,34 @@ export class UserService {
     return this.userRepo.save({ ...user, ...update });
   }
 
-  async deactivateUser(id: number, allUser = false): Promise<void> {
+  async updateAddress(id: number, address: string, dto: UpdateAddressDto): Promise<{ user: UserV2Dto }> {
+    const userData = await this.userDataRepo.findOne({
+      where: { id },
+      relations: { users: { wallet: true } },
+    });
+    if (!userData) throw new NotFoundException('User not found');
+
+    const userToUpdate = userData.users.find((u) => u.address === address);
+    if (!userToUpdate) throw new NotFoundException('Address not found');
+
+    userToUpdate.label = dto.label;
+    await this.userRepo.save(userToUpdate);
+
+    return { user: UserDtoMapper.mapUser(userData) };
+  }
+
+  async deactivateUser(id: number, address?: string): Promise<void> {
     const mainUser = await this.userRepo.findOne({ where: { id }, relations: { userData: { users: true } } });
     if (!mainUser) throw new NotFoundException('User not found');
     if (mainUser.userData.isBlocked || mainUser.userData.isDeactivated)
       throw new BadRequestException('User Account already deactivated');
-    if (mainUser.isBlockedOrDeactivated) throw new BadRequestException('User already deactivated');
 
-    if (!allUser) {
-      await this.userRepo.update(...mainUser.deactivateUser('Manual user deactivation'));
+    if (address) {
+      const user = mainUser.userData.users.find((u) => u.address === address);
+      if (!user) throw new NotFoundException('User not found');
+      if (user.status === UserStatus.BLOCKED) throw new BadRequestException('User already deactivated');
+
+      await this.userRepo.update(...user.deactivateUser('Manual user account block'));
```

Comment:
Change to `Manual user deactivation` from dev changes instead of `Manual user account block`